### DISCUSSION TITLE
AWS Deployment Strategies

### DISCUSSION THREAD

**Discussion by Kevin-Luke** [Left4DevOps/l4d2-docker#10] [Ideas]
In regards to the recent issue opened by me, I thought it would be helpful to discuss some out of box deployment solutions which might be added to the repository as examples:

My first two strategies were similar to yours.

1. Create a terraform solution that a creates a basic ec2 instance and uses a cloud-init script to manage the docker process via systemd. Solution includes a basic security group and targets a public subnet. This would allow someone to spin up an instance ad-hoc to quickly get a random pub IP from AWS and play with their friends without having to port forward and poke holes in their local. Average cost would be about 4 to 8 cents an hour for about 2 to 4 hours play time, then they could simply terraform destroy the whole thing.

2. Terraform ECS deployment. Same as above except container native. Since this is a single container, this seems like the optimal solution.  Additions could include building your own container with mods and pushing it to ECR.

I'd be happy to open a PR with one of these as an example when I have it prepared if it would be beneficial.

**Comment by hazmeister** [comment]
Hi Kevin, thanks for the suggestions. I've cleaned up my own deployment and stuck it a new repo.  It still needs some more documentation and testing, but you may find it interesting.

https://github.com/Left4DevOps/l4d2-terraform
